services/spar/src/Spar/App.hs

```diff
@@ -350,19 +351,13 @@ catchVerdictErrors = (`catch` hndlr)
   where
     hndlr :: SparError -> Sem r VerdictHandlerResult
     hndlr err = do
-      waiErr <- renderSparErrorWithLogging err
-      pure $ case waiErr of
-        Right (werr :: Wai.Error) ->
-          VerifyHandlerError
-            (LText.toStrict $ Wai.label werr)
-            (LText.toStrict $ Wai.message werr)
-        Left (serr :: ServerError) ->
-          VerifyHandlerError
-            "unknown-error"
-            ( Text.pack (errReasonPhrase serr)
-                <> " "
-                <> (Text.decodeUtf8With lenientDecode . toStrict . errBody $ serr)
-            )
+      werr <- renderSparErrorWithLogging err <&> httpErrorToWaiError
+      -- TODO: we don't want to include the RichError part of
```

Comment:
Was staring at the `RichError` data construction as well and wondering when if ever is the rich part used. It seems it's currently used in the Eq instance (newly added in this PR), and then constructed/used here `services/brig/src/Brig/API/Error.hs:203:deleteUserError (DeleteUserPendingCode t) = RichError deletionCodePending (DeletionCodeTimeout t) []`, everywhere else it's filled with `()`/`Null`. I guess the question is: what is it's purpose, internal for debugging or could the outside world see it?